src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -454,6 +454,17 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     }
   }
 
+  def listClustersByGoogleProject(userInfo: UserInfo, googleProject: GoogleProject, params: LabelMap): Future[Seq[Cluster]] = {
+    for {
+      paramMap <- processListClustersParameters(params)
+      clusterList <- dbRef.inTransaction { da => da.clusterQuery.listByLabelsByProject(googleProject, paramMap._1, paramMap._2) }
+      visibleClusters <- authProvider.filterUserVisibleClusters(userInfo, clusterList.map(c => (c.googleProject, c.clusterName)).toList)
```

Comment:
we are duplicating most of the code from `listClusters`, wondering if we can refactor to reduce duplication.

Reply:
Yeah maybe.. will try.

Reply:
One option is to modify `def listClusters` to take in an optional GoogleProject:

```scala
def listClusters(userInfo: UserInfo, params: LabelMap, googleProject: Option[GoogleProject] = None)
```

and then call `listClusters` or `listClustersByGoogleProject` based on pattern matching. I think we could further refactor this by doing something similar with `listByLabels` where it also accepts an optional project and filter or not filter based on its value. That way `LeonardoService.listClusters` would just pass down the optional parameter.

Reply:
I did this (passing the `Option[GoogleProject]` downt he `ClusterComponent`)